front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap/RenderBlockState.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Source } from 'react-map-gl';
+import lineSliceAlong from '@turf/line-slice-along';
+import { Feature, LineString } from 'geojson';
+import { RootState } from 'reducers';
+import OrderedLayer from 'common/Map/Layers/OrderedLayer';
+import { TrainPosition } from './types';
+
+function setFirstBlockState(
+  geojsonPath: Feature<LineString>,
+  headDistance: number,
+  tailDistance: number
+) {
+  const threshold = 0;
+  if (headDistance - tailDistance > threshold) {
+    return lineSliceAlong(geojsonPath, tailDistance, headDistance);
+  }
+  if (headDistance > threshold) {
+    return lineSliceAlong(geojsonPath, headDistance - threshold, headDistance);
+  }
+  return lineSliceAlong(geojsonPath, 0, threshold);
+}
+
+function setSecondBlockState(
+  geojsonPath: Feature<LineString>,
+  headDistance: number,
+  tailDistance: number
+) {
+  const threshold = 0;
+  if (headDistance - tailDistance > threshold) {
+    return lineSliceAlong(geojsonPath, tailDistance, headDistance);
+  }
+  if (headDistance > threshold) {
+    return lineSliceAlong(geojsonPath, headDistance - threshold, headDistance);
+  }
+  return lineSliceAlong(geojsonPath, 0, threshold);
+}
+
+interface RenderBlockStateProps {
+  point: TrainPosition;
+  geojsonPath: Feature<LineString>;
+  layerOrder: number;
+}
+
+export default function RenderBlockState(props: RenderBlockStateProps) {
+  const { point, geojsonPath, layerOrder } = props;
+  const { selectedTrain } = useSelector((state: RootState) => state.osrdsimulation);
+  const simulation = useSelector((state: RootState) => state.osrdsimulation.simulation.present);
+  const data = simulation.trains[selectedTrain].base.route_aspects;
+
+  function getFirstBlockState(geojsonPath: Feature<LineString>, number1: number, number2: number) {
```

Comment:
`number1` is meaningless, name your variables with explicit names of what it represents not what it contains (the type is enough !)